### PR TITLE
Add CryptoWatchPriceFeed

### PR DESCRIPTION
**Pull request by mrice32**
This adds a price feed based on the cryptowatch api: https://docs.cryptowat.ch/rest-api/.

It also adds a `Networker` object that enables mocking of network calls/responses in the tests.

Next step: construct a price feed that aggregates multiple exchange price feeds.

Note: an aggregated price feed will just compose the outputs of multiple individual price feeds by taking them in its constructor.

### CODE REVIEW

**Comment by mrice32** [comment]
@ptare when you get a chance, could you take a look at this one?

File: financial-templates-lib/price-feed/CryptoWatchPriceFeed.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  // exchange a string identifier for the echange to pull prices from. This should be the identifier used to
+  //          identify the exchange in CW's REST API.
+  // pair a string representation of the pair the price feed is tracking. This pair should be available on the
+  //      provided exchange. The string should be the representation used by CW to identify this pair.
+  // lookback how far in the past the historical prices will be available using the getHistoricalPrice function.
+  // networker networker object used to send the API requests.
+  // getTime function to return the current time.
+  // minTimeBetweenUpdates min number of seconds between updates. If update() is called again before this number of
+  // seconds has passed, it will be a no-op.
+  constructor(web3, apiKey, exchange, pair, lookback, networker, getTime, minTimeBetweenUpdates) {
```
Comment (chrismaree) [inline financial-templates-lib/price-feed/CryptoWatchPriceFeed.js:16]: This looks great but should we not ingest a logger like with our other infrastructure to emit events on updates and throw errors when something breaks?

File: financial-templates-lib/price-feed/CryptoWatchPriceFeed.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+  getLastUpdateTime() {
+    return this.lastUpdateTime;
+  }
+
+  async update() {
+    const { toWei, toBN } = this.web3.utils;
+    const currentTime = this.getTime();
+
+    // Return eatly if the last call was too recent.
```
Comment (ptare) [inline financial-templates-lib/price-feed/CryptoWatchPriceFeed.js:69]: typo: eatly

File: financial-templates-lib/price-feed/Networker.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+// This class makes networking calls on behalf of the caller.
+// Note: this is separated out to allow this functionality to be mocked out in tests so no real network calls have to
+// be made.
+class Networker {
+  async getJson(url) {
+    const response = await fetch(url);
+    const json = await response.json();
+    if (!json) {
+      throw `Query [${url}] failed to get JSON`;
```
Comment (ptare) [inline financial-templates-lib/price-feed/Networker.js:11]: use the logger?

File: financial-templates-lib/test/price-feed/NetworkerMock.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// An implementation of PriceFeedInterface that uses a Uniswap v2 TWAP as the price feed source.
```
Comment (ptare) [inline financial-templates-lib/test/price-feed/NetworkerMock.js:1]: wrong comment

File: financial-templates-lib/price-feed/CryptoWatchPriceFeed.js (hunk 1 above)
Comment (mrice32) [reply financial-templates-lib/price-feed/CryptoWatchPriceFeed.js]: Done. PTAL.

File: financial-templates-lib/price-feed/CryptoWatchPriceFeed.js (hunk 2 above)
Comment (mrice32) [reply financial-templates-lib/price-feed/CryptoWatchPriceFeed.js]: Done.

File: financial-templates-lib/price-feed/Networker.js (hunk 3 above)
Comment (mrice32) [reply financial-templates-lib/price-feed/Networker.js]: Done. PTAL.

File: financial-templates-lib/test/price-feed/NetworkerMock.js (hunk 4 above)
Comment (mrice32) [reply financial-templates-lib/test/price-feed/NetworkerMock.js]: Fixed.

**Pull request merged**